document-store/src/test/java/org/hypertrace/core/documentstore/FilterTest.java

```diff
@@ -61,4 +62,16 @@ public void testSettingChildFiltersField() {
     Assertions.assertNotNull(filter2.getChildFilters());
     Assertions.assertEquals(0, filter2.getChildFilters().length);
   }
+
+  @Test
+  public void testEquals() {
+    Filter filter1 = Filter.eq("entity_type", "SERVICE");
```

Comment:
nit: It's better to do simple checks like `filter1.equals(filter1)` and `!filter1.equals(filter2)`

Reply:
Didn't check for `filter1.equals(filter1)` because it passes the trivial check `filter1 == filter1`.